nifi-toolkit/nifi-toolkit-encrypt-config/src/main/groovy/org/apache/nifi/toolkit/encryptconfig/DecryptMode.groovy

```diff
@@ -0,0 +1,322 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.nifi.toolkit.encryptconfig
+
+import org.apache.commons.cli.HelpFormatter
+import org.apache.nifi.properties.AESSensitivePropertyProvider
+import org.apache.nifi.properties.SensitivePropertyProvider
+import org.apache.nifi.toolkit.encryptconfig.util.BootstrapUtil
+import org.apache.nifi.toolkit.encryptconfig.util.PropertiesEncryptor
+import org.apache.nifi.toolkit.encryptconfig.util.ToolUtilities
+import org.apache.nifi.toolkit.encryptconfig.util.XmlEncryptor
+import org.apache.nifi.util.console.TextDevices
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+
+class DecryptMode implements ToolMode {
+
+    private static final Logger logger = LoggerFactory.getLogger(DecryptMode.class)
+
+    enum FileType {
+        properties,
+        xml
+    }
+
+    CliBuilder cli
+
+    DecryptMode() {
+        cli = cliBuilder()
+    }
+
+    void printUsage(String message = "") {
+        if (message) {
+            System.out.println(message)
+            System.out.println()
+        }
+        cli.usage()
+    }
+
+    void printUsageAndExit(String message = "", int exitStatusCode) {
+        printUsage(message)
+        System.exit(exitStatusCode)
+    }
+
+    @Override
+    void run(String[] args) {
+        logger.warn("The decryption capability of this tool is still considered experimental. The results should be manually verified.")
+        try {
+
+            def options = cli.parse(args)
+
+            if (!options || options.h) {
+                printUsageAndExit("", EncryptConfigMain.EXIT_STATUS_OTHER)
+            }
+
+            EncryptConfigLogger.configureLogger(options.v)
+
+            DecryptConfiguration config = new DecryptConfiguration(options)
+
+            run(config)
+
+        } catch (Exception e) {
+            logger.error("Encountered an error: ${e.getMessage()}")
+            logger.debug("", e) // stack trace only when verbose enabled
```

Comment:
I understand why this was done but I think the better logic is:

```
if (isVerboseEnabled()) {
    logger.error("", e)
}
```
Some people/tools are set up to extract messages based on levels, and this is semantically an error.

Reply:
Good call.
- [ ] Change error logging to got to error, wrap in conditional if it depends on verbose